src/main/webapp/dev/js/common/questionRubric.js

```diff
@@ -367,6 +379,7 @@ function toggleAssignWeightsRow($checkbox) {
     if ($checkbox.prop('checked')) {
         $weightsRow.show();
         $weightsRowFirstCell.html('Weights <span class="glyphicon glyphicon-arrow-right"></span>');
+        makeRubricWeightsRequired($weightsRow);
     } else {
         $weightsRow.hide();
```

Comment:
Should't we remove the required field? We should not depend on the `removeRequiredIfElementHidden` to remove things for us.

Reply:
@xpdavid But doesn't the `required` field only validate if the element is visible? The `removeRequiredIfElementHidden` is a temporary solution for a bug in chrome browser only (https://github.com/TEAMMATES/teammates/issues/8688#issuecomment-375134130, https://github.com/TEAMMATES/teammates/pull/8678#discussion_r175628789). And for that bug, I have to re-add the `required` using this method.

I was wondering then, is there any reason for removing the `required` field here, instead of depending on the `removeRequiredIfElementHidden` method?

Reply:
Ok. I believe it make more sense to add `makeRubricWeightsRequired` to `enableNewQuestion`. It has nothing to do with the toggle logic.

Reply:
>I believe it make more sense to add `makeRubricWeightsRequired` to `prepareQuestionForm`. It has nothing to do with the toggle logic.

@xpdavid Tried that, Turns out removing the `makeRubricWeightsRequired` from the toggle function doesn't solve the bug I mentioned in https://github.com/TEAMMATES/teammates/pull/8863#discussion_r192446598. Actually, what is happening is, the `removeRequired*` method gets triggered only when we click on the add question button. And the problem I mentioned in the above mentioned comment will occur once the new question form is ready, and The user has tried to add the question and failed in a validation (like in empty text case). In this case, `prepareQuestionForm` doesn't get triggered again, so the `required` field in rubric weight cells are not added as we would expect.

Reply:
@xpdavid Based on https://github.com/TEAMMATES/teammates/pull/8863#discussion_r192558337, should I move `removeRequiredIfElementHidden();` in `showNewQuestionFrame` or should I leave it as it is?

Reply:
1. In `removeRequiredIfElementHidden`, we are doing event binding. It extremely weird to call the event binding again and again when the `showNewQuestionFrame` is called (which is adding new question) (Image that we cancel, add, cancel, add again and again. We should feel luck that JQuery doesn't complain).

2. I don't think it is browser's faults of validating required element which is hidden. It is developer's faults at all. Therefore, it is necessary to remove the required field when the checkbox is unchecked.

Reply:
>Therefore, it is necessary to remove the required field when the checkbox is unchecked.

Okay, modifying to make sure the `required` field is removed when checkbox is unchecked.